build.gradle

```diff
@@ -971,6 +976,38 @@ idea {
                     ]
                 }
             }
+            if (ideaOverrideBuildType != "") {
+                // only modify the build type if user has expressed a desire for change
+                withIDEAFileXml("gradle.xml") {
+                    def gradleSettings = it.asNode()
+                        .findFirstOrAppendChild('component', ['name': 'GradleSettings'])
+                        .findFirstOrAppendChild('option', ['name': 'linkedExternalProjectsSettings'])
+                        .findFirstOrAppendChild('GradleProjectSettings')
+
+                    def options
+                    if ("gradle".equalsIgnoreCase(ideaOverrideBuildType))
+                        options = ['delegatedBuild': 'true', 'testRunner': 'GRADLE']
+                    else if ("idea".equalsIgnoreCase(ideaOverrideBuildType))
+                        options = ['delegatedBuild': 'false', 'testRunner': 'PLATFORM']
+                    else
+                        throw GradleScriptException('Accepted value for ideaOverrideBuildType is one of gradle or idea.')
+
+                    options.each { k, v ->
+                        gradleSettings.findFirstOrAppendChild('option', ['name': k]).attributes().value = v
+                    }
+                }
+            }
+            withIDEAFileXml("misc.xml") {
```

Comment:
this is something that needs further testing with the three IJ configurations: folder-based, file-based and implicit (no files) projects imported from gradle, iirc the xml functions were causing me trouble in one of these

Reply:
i would assume everyone is using imported from gradle, no? And if they don’t import from gradle this won’t run at all right?

Reply:
I meant this thing, though apparently file-based projects are now deprecated, I didn't realise it: <https://www.jetbrains.com/help/idea/creating-and-managing-projects.html#file-based>